### PR TITLE
Improve documentation and add warnings re: params-file vs configuration

### PR DESCRIPTION
**Pull request by jfy133**
Improves documentation about this issue: https://github.com/nextflow-io/nextflow/issues/2662

### CODE REVIEW

File: docs/cli.rst [hunk 1]
```diff
@@ -1229,10 +1229,21 @@ facilitates rapid iterations, inspections of any pipeline as well as debugging.
     $ nextflow run nextflow-io/hello -with-tower
 
 - Invoke the nextflow pipeline execution with a custom parameters ``YAML/JSON`` file.
-  The parameters which are specified through this mechanism are merged with the resolved configuration (base configuration and profiles) and only the common fields are overwritten by the ``YAML/JSON`` file.::
+  The parameters which are specified through this mechanism are merged with the resolved configuration (base configuration and profiles) and only the common fields are overwritten by the ``YAML/JSON`` file. ::
 
     $ nextflow run main.nf -params-file pipeline_params.yml
 
+  For example, the following params file::
```
Comment (bentsherman) [inline docs/cli.rst:1236]:
```suggestion
  For example, the following params file (in JSON)::
```

**Review by bentsherman — APPROVED** [review]

**Comment by jfy133** [comment]
@bentsherman I guess I can't fix the DCO error correct?

File: docs/cli.rst [hunk 2]
```diff
@@ -1229,10 +1229,21 @@ facilitates rapid iterations, inspections of any pipeline as well as debugging.
...
+
+    {
+      "alpha": 1,
+      "beta": "foo"
+    }
+
+  Is equivalent to the following command line::
+
+    $ nextflow run main.nf --alpha 1 --beta foo
+
```
Comment (jfy133) [inline docs/cli.rst:1246]: An example of a params file in YAML format::

    alpha: 1
    beta: 'foo'

Comment (jfy133) [reply docs/cli.rst:1246]: One further addition

File: docs/cli.rst [hunk 3]
```diff
@@ -1229,10 +1229,21 @@ facilitates rapid iterations, inspections of any pipeline as well as debugging.
...
+  The parameters which are specified through this mechanism are merged with the resolved configuration (base configuration and profiles) and only the common fields are overwritten by the ``YAML/JSON`` file. ::
 
     $ nextflow run main.nf -params-file pipeline_params.yml
 
+  For example, the following params file (in JSON)::
+
+    {
+      "alpha": 1,
+      "beta": "foo"
+    }
```
Comment (bentsherman) [inline docs/cli.rst:1241]:
```suggestion
  For example, the following params file in YAML format::

    alpha: 1
    beta: 'foo'
```

**Comment by bentsherman** [comment]
Hmm, I don't know why the DCO isn't working for my commits. Maybe because I'm committing through GitHub. You could roll everything back on your local machine, make one big commit, and force push it with your sign-off. Otherwise Paolo might be able to fix it.

**Comment by jfy133** [comment]
Ok, didn't get this to work properly, will start with a new PR (sorry :face_exhaling: )

**Pull request closed without merging**